### PR TITLE
Add meson support

### PR DESCRIPTION
**Pull request by modbw**
Initial support for meson based  build of socketcand.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,31 @@
+project('socketcand', 'c', version : '0.6.1')
+conf = configuration_data()
+
+deps = [dependency('threads')]
+
+# Check if libconfig is available
```
Comment (yegorich) [inline meson.build:6]: Could you create a `meson_options.txt` and make this dependency as a boolean option like this:

`option('libconfig', type: 'boolean', value: false, description: 'enable support for libconfig')`

And then in `meson.build`:

```
enable_libconfig = get_option('libconfig')
if enable_libconfig
  dep_libconfig = dependency('libconfig', version: '>= 1.7.3')
endif
```

**Comment by modbw** [comment]
meson_options.txt and socketcandcl added

File: meson.build (hunk 1 above)
Comment (modbw) [reply meson.build:6]: I have done so. Please review.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+enable_libconfig = get_option('libconfig')
+if enable_libconfig
+    dep_libconfig = dependency('libconfig', version: '>= 1.7.3')
```
Comment (yegorich) [inline meson.build:9]: Please, remove the version constraint. I have `libconfig` 1.5 and `socketcand` compiles and links without any problem. I think, the project requires the very basic `libconfig` funcionality. Hence, don't ask for the special version.

Other than that, everything looks fine.

Comment (modbw) [reply meson.build]: Done.

**Comment by yegorich** [comment]
The PR is ready for merging. @marckleinebudde any comments?

**Comment by yegorich** [comment]
I'll take care of this. The first PR updating the `README.md` is out.

**Comment by marckleinebudde** [comment]
Can you also take care of:

```
init.d/socketcand.in
rc.d/socketcand
etc/socketcand.conf
socketcand.1
```

**Comment by yegorich** [comment]
Will do

**Comment by marckleinebudde** [comment]
Just noticed, there is also:
```
debian_pack.in
```

**Comment by yegorich** [comment]
I have also seen this file. From my point of view, there are two possibilites:

- port `debian_pack.in` to meson
- create the standard Debian infrastructure like it is done in [Timeshift](https://github.com/yegorich/timeshift). See the `debian` folder

I prefer the latter. This way you can even create packages for other archs using the standard cross-compiler or [ELBE](https://elbe-rfs.org/).

What do you think?

**Comment by marckleinebudde** [comment]
> I prefer the latter. This way you can even create packages for other archs using the standard cross-compiler or [ELBE](https://elbe-rfs.org/).

Make it so!

**Pull request merged**